docs/examples/vault-support/README.md

```diff
@@ -0,0 +1,71 @@
+# HashiCorp Vault Support Example
+
+The RabbitMQ Cluster Operator supports storing RabbitMQ default user (admin)
+credentials and RabbitMQ server certificates in
+[HashiCorp Vault](https://www.vaultproject.io/) and in K8 Secrets. And
+accordingly, the RabbitMQ Messaging Topology operator can operate on clusters
+which are vault-enabled or k8s-secrets enabled.
+
+As explained in [this KubeCon talk](https://youtu.be/w0k7MI6sCJg?t=177) there
+are four different approaches in Kubernetes to consume external secrets:
+
+1. Direct API
+2. Controller to mirrors secrets in K8s
+3. Sidecar + MutatingWebhookConfiguration
+4. Secrets Store CSI Driver
+
+The RabbitMQ Cluster Operator integrates with Vault using the third approach
+(Sidecar + MutatingWebhookConfiguration) whereas the RabbitMQ Messaging
+Topology Operator uses the first approach (Direct API).
+
+## Vault-related configuration required
+
+The Vault server must have the version 2 key value secret engine and the 
+[Vault Kubernetes auth method](https://www.vaultproject.io/docs/auth/kubernetes)
+enabled.
+
+```bash
+$ vault secrets enable -path=secret kv-v2
+```
+
+```bash
+$ vault auth enable kubernetes
+```
+
+In order for the RabbitMQ Messaging Topology operator to authenticate with
+a Vault server and access RabbitMQ cluster default user credentials it is
+necessary for the operator container to have one or more environment variables
+set. 
+
+- `VAULT_ADDR` should be set to the URL of the Vault server API
+- `OPERATOR_VAULT_ROLE` should be set to the name of the Vault role that is used when accessing credentials. This defaults to "messaging-topology-operator"
+- `OPERATOR_VAULT_NAMESPACE` may be set to the Vault namespace to use when the Messaging Topology operator is authenticating. If not set then the default Vault namespace is assumed
```

Comment:
Minor suggestion:  add a link to the [Vault namespace](https://www.vaultproject.io/docs/enterprise/namespaces) so that it is not confused with kubernetes namespaces.